### DISCUSSION TITLE
自建服务器无法显示弹幕

### DISCUSSION THREAD

**Discussion by MuYu-Blog** [xfgryujk/blivechat#215] [提问]
启动报错

024-07-04 01:26:13,276 INFO [__main__]: App started, initializing 2024-07-04 01:26:13,291 INFO [__main__]: Server started: 127.0.0.1:12450 2024-07-04 01:26:13,291 INFO [__main__]: Running event loop 2024-07-04 01:26:13,293 INFO [services.translate]: TencentTranslateFree waiting to become available 2024-07-04 01:26:13,346 ERROR [services.translate]: TencentTranslateFree init error: Traceback (most recent call last): File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/connector.py", line 1025, in _wrap_create_connection return await self._loop.create_connection(*args, **kwargs) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/asyncio/base_events.py", line 1149, in create_connection transport, protocol = await self._create_connection_transport( ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/asyncio/base_events.py", line 1182, in _create_connection_transport await waiter File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/asyncio/sslproto.py", line 578, in _on_handshake_complete raise handshake_exc File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/asyncio/sslproto.py", line 560, in _do_handshake self._sslobj.do_handshake() File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/ssl.py", line 917, in do_handshake self._sslobj.do_handshake() ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1000)  The above exception was the direct cause of the following exception:  Traceback (most recent call last): File "/www/wwwroot/Blivechat/services/translate.py", line 327, in _do_init async with utils.request.http_session.get('https://fanyi.qq.com/') as r: File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/client.py", line 1197, in __aenter__ self._resp = await self._coro ^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/client.py", line 581, in _request conn = await self._connector.connect( ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/connector.py", line 544, in connect proto = await self._create_connection(req, traces, timeout) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/connector.py", line 944, in _create_connection _, proto = await self._create_direct_connection(req, traces, timeout) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/connector.py", line 1257, in _create_direct_connection raise last_exc File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/connector.py", line 1226, in _create_direct_connection transp, proto = await self._wrap_create_connection( ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/www/server/pyporject_evn/Blivechat_venv/lib/python3.12/site-packages/aiohttp/connector.py", line 1027, in _wrap_create_connection raise ClientConnectorCertificateError(req.connection_key, exc) from exc aiohttp.client_exceptions.ClientConnectorCertificateError: Cannot connect to host fanyi.qq.com:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1000)')] Failed to check update: connection failed

![image](https://github.com/xfgryujk/blivechat/assets/65322181/d7807150-ae51-42ea-bb83-a7a9e92213a9)
无法显示弹幕

**Comment by xfgryujk** [comment]
你没装证书？

**Reply by MuYu-Blog** [reply]
SSL证书嘛 
我是通过宝塔面板的Python网站项目部署的
在里面已经添加了现有该域名的SSL证书
![image](https://github.com/xfgryujk/blivechat/assets/65322181/a2b8e3df-f875-4043-b1a6-38f122aa2fe0)
![image](https://github.com/xfgryujk/blivechat/assets/65322181/7fac5a2b-c859-4434-b31d-aa0438af46f6)
![image](https://github.com/xfgryujk/blivechat/assets/65322181/1ab68176-9bc2-452a-9a32-bbe6280e1858)

**Reply by xfgryujk** [reply]
不知道你开了什么奇怪的代理，反正 it works on my machine

**Reply by MuYu-Blog** [reply]
奇怪的问题(O_O) 并没有开任何代理
